### PR TITLE
remove broadcast_socket from main library

### PR DESCRIPTION
**Pull request by arvidn**
it's only used by `test_upnp` now

### CODE REVIEW

File: include/libtorrent/address.hpp [hunk 1]
```diff
@@ -73,6 +73,29 @@ namespace libtorrent {
 	using boost::asio::ip::make_address_v4;
 	using boost::asio::ip::make_address_v6;
 #endif
+
+	TORRENT_EXTRA_EXPORT bool is_local(address const& a);
```
Comment (aldenml) [inline include/libtorrent/address.hpp]: what do you think about putting these inside `aux_/address.hpp` instead?

Comment (arvidn) [reply include/libtorrent/address.hpp]: yeah, that probably makes sense. If I'm adding a new header, I might as well call it something else too.

**Comment by arvidn** [comment]
@aldenml does this look reasonable?

**Comment by aldenml** [comment]
Sometimes I do wonder if using `aux::` all over the place (like `std::`) instead of using `using namespace aux` is the right approach, when done incrementally it goes unnoticed, but it's adding up.

**Comment by arvidn** [comment]
I'm thinking that once all the internal code is actually *defined* in the `aux` namespace, a lot of it could be removed

**Pull request merged**